pkg/migrator/errors.go

```diff
@@ -52,6 +53,18 @@ func isConnectionRefusedError(err error) bool {
 	return strings.Contains(err.Error(), "connection refused")
 }
 
+// isUIDPreconditionError checks whether the error contains "Precondition failed: UID in precondition".
+// This error message is typically present when the resource is deleted before it can be written back.
+func isUIDPreconditionError(err error) bool {
+	return strings.Contains(err.Error(), "Precondition failed: UID in precondition")
+}
+
+// isConflictError checks whether the given error is a conflict error.
+// This is true when errors.IsConflict or isUIDPreconditionError returns true.
+func isConflictError(err error) bool {
+	return errors.IsConflict(err) || isUIDPreconditionError(err)
+}
```

Comment:
If the UID precondition fails, doesn't that imply that the current iteration of the resource was created after the migration started? And if so, is it redundant to "migrate" it?

Reply:
I think that this issue was originally raised because the migrator was trying to migrate resources that were deleted during migration, so there was a race. So the precondition failed with the stored UID being empty.

Reply:
Don't API servers generally return HTTP 404 when that happens? How can it enforce a UID precondition at all if the named object does not exist?